domain/schema/model/sql/0017-machine.sql

```diff
@@ -27,6 +27,18 @@ ON machine (name);
 CREATE UNIQUE INDEX idx_machine_net_node
 ON machine (net_node_uuid);
 
+CREATE TABLE machine_parent (
+    machine_uuid TEXT NOT NULL PRIMARY KEY,
+    parent_uuid TEXT NOT NULL PRIMARY KEY,
+
+    CONSTRAINT fk_machine_parent_machine
+    FOREIGN KEY (machine_uuid)
+    REFERENCES machine (uuid),
+    CONSTRAINT fk_machine_parent_parent
+    FOREIGN KEY (parent_uuid)
+    REFERENCES machine (uuid)
+);
```

Comment:
We've been going with a singular `PRIMARY KEY` to make it easier to understand.

```suggestion
CREATE TABLE machine_parent (
    machine_uuid TEXT NOT NULL,
    parent_uuid TEXT NOT NULL,
    CONSTRAINT fk_machine_parent_machine
    FOREIGN KEY (machine_uuid)
    REFERENCES machine (uuid),
    CONSTRAINT fk_machine_parent_parent
    FOREIGN KEY (parent_uuid)
    REFERENCES machine (uuid),
    PRIMARY KEY (machine_uuid, parent_uuid)
);
```

Reply:
Yes seems good, thanks !